pkg/common/plugin/download.go

```diff
@@ -0,0 +1,111 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore/oras/authprovider"
```

Comment:
Would like to propose moving `authprovider` to a common location in a standalone refactor PR, then consume in both places

Reply:
PR per consensus on community call here: https://github.com/deislabs/ratify/pull/520